share/functions/fish_job_summary.fish

```diff
@@ -29,6 +29,8 @@ function fish_job_summary -a job_id is_foreground cmd_line signal_or_end_name si
 
     set -l message
     switch $signal_or_end_name
+        case STATUS
+            set message (printf ( _ "fish: Job %s, '%s' has Non Zero status code:%s\n" ) $job_id $cmd_line $signal_desc)
```

Comment:
As `$signal_desc` may now be either the exit code or the status value, please update the "docstring" at the top of the function, and rename the variable.

As for the message, I suggest just extending the ENDED message:

```
set message (printf ( _ "fish: Job %s, '%s' has ended with status %d\n" ) $job_id $cmd_line $status_num)
```